src/misc/api_trace.cc

```diff
@@ -0,0 +1,418 @@
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclAllToAllv_fn,                  3);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclBroadcast_fn,                  4);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGather_fn,                     5);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclReduce_fn,                     6);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclReduceScatter_fn,              7);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclScatter_fn,                    8);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclSend_fn,                       9);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclRecv_fn,                       10);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclRedOpCreatePreMulSum_fn,       11);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclRedOpDestroy_fn,               12);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGroupStart_fn,                 13);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGroupEnd_fn,                   14);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGetVersion_fn,                 15);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGetUniqueId_fn,                16);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommInitRank_fn,               17);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommInitAll_fn,                18);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommInitRankConfig_fn,         19);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommFinalize_fn,               20);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommDestroy_fn,                21);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommAbort_fn,                  22);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommSplit_fn,                  23);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGetErrorString_fn,             24);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclGetLastError_fn,               25);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommGetAsyncError_fn,          26);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommCount_fn,                  27);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommCuDevice_fn,               28);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommUserRank_fn,               29);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclMemAlloc_fn,                   30);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, mscclLoadAlgo_fn,                  31);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, mscclRunAlgo_fn,                   32);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, mscclUnloadAlgo_fn,                33);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommRegister_fn,               34);
+    RCCL_ASSERT_OFFSET(rcclApiFuncTable, ncclCommDeregister_fn,             35);
+
+
+#undef RCCL_ASSERT_OFFSET
+
+    static_assert(
+        sizeof(rcclApiFuncTable) == compute_table_size(36),
+        "Update table major/step version and add a new offset assertion if this fails to compile");
+
+    static std::array<unsigned char, sizeof(rcclApiFuncTable)> m_buffer;
```

Comment:
```suggestion
    std::array<unsigned char, sizeof(rcclApiFuncTable)> m_buffer;
```

`static` is redundant in an anonymous namespace